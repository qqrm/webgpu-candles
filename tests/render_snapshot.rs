use wasm_bindgen::JsCast;
use wasm_bindgen_test::*;

wasm_bindgen_test_configure!(run_in_browser);

#[wasm_bindgen_test]
fn red_canvas_snapshot() {
    let window = web_sys::window().unwrap();
    let document = window.document().unwrap();
    let canvas = document
        .create_element("canvas")
        .unwrap()
        .dyn_into::<web_sys::HtmlCanvasElement>()
        .unwrap();
    canvas.set_width(10);
    canvas.set_height(10);
    document.body().unwrap().append_child(&canvas).unwrap();

    let ctx = canvas
        .get_context("2d")
        .unwrap()
        .unwrap()
        .dyn_into::<web_sys::CanvasRenderingContext2d>()
        .unwrap();
    ctx.set_fill_style_str("#ff0000");
    ctx.fill_rect(0.0, 0.0, 10.0, 10.0);

    let data_url = canvas.to_data_url().unwrap();
    insta::assert_snapshot!(data_url);
}
