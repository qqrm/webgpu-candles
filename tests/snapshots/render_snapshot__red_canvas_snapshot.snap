---
source: tests/render_snapshot.rs
expression: data_url
---
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAKCAYAAACNMs+9AAAAGElEQVR4nGP8z8Dwn4EIwESMolGF1FMIAD2cAhK2AyPVAAAAAElFTkSuQmCC
